pyproject.toml

```diff
@@ -0,0 +1,72 @@
+[build-system]
+requires = ["setuptools>=64", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "joinmarket"
+version = "0.9.11dev"
+description = "Joinmarket client library for Bitcoin coinjoins"
+readme = "README.md"
+requires-python = ">=3.7"
+license = {file = "LICENSE"}
+dependencies = [
+    "chromalog==1.0.5",
+    "pyaes==1.6.1",
+    "service-identity==21.1.0",
+    "txtorcon==22.0.0",
+    "twisted==22.4.0",
+]
+
+[project.optional-dependencies]
+jmbitcoin = [
+    "python-bitcointx==1.1.3",
+]
+jmclient = [
+    "argon2_cffi==21.3.0",
+    "autobahn==20.12.3",
+    "bencoder.pyx==3.0.1",
+    "klein==20.6.0",
+    "mnemonic==0.20",
+    "pyjwt==2.4.0",
+    "werkzeug==2.2.3",
+]
+jmdeamon = [
+    "libnacl==1.8.0",
+    "pyopenssl==23.2.0",
+    'cryptography==3.3.2; platform_machine != "aarch64" and platform_machine != "amd64" and platform_machine != "x86_64"',
+    'cryptography==41.0.2; platform_machine == "aarch64" or platform_machine == "amd64" or platform_machine == "x86_64"',
+]
+jmqtui = [ "PyQt5!=5.15.0,!=5.15.1,!=5.15.2,!=6.0" ]
+client = [
+    "joinmarket[jmclient]",
+    "joinmarket[jmbitcoin]",
+]
+deamon = [
+    "joinmarket[jmdeamon]",
+]
+services = [
```

Comment:
I understand and agree with this structure of dependencies, but, and I'm probably just being dumb here, why isn't `jmbase` listed here? or is it somewhere explicit that `joinmarket` itself requires `jmbase`?